### PR TITLE
Fix HealthChecker to log dependency exceptions

### PR DESCRIPTION
**Pull request by johnlcox**
What was changed? Why is this necessary?
----------------------------------------

Fix `HealthChecker` to actually log dependency exceptions. Previously it was passing the `Optional` to the logger rather than the actual exception.

How was it tested?
----

Existing unit/integration tests.

How to test
----

*This is bare minimum acceptable testing*

- [ ] `mvn clean install -U`

### CODE REVIEW

File: health/service/src/main/java/com/cerner/beadledom/health/internal/HealthChecker.java [hunk 1]
```diff
@@ -31,6 +31,7 @@
  */
 public class HealthChecker {
```
Comment (nathanschile) [inline health/service/src/main/java/com/cerner/beadledom/health/internal/HealthChecker.java]: Does a test need to be updated to test that your changes work?

Comment (johnlcox) [reply health/service/src/main/java/com/cerner/beadledom/health/internal/HealthChecker.java]: How can we test loggers?

Comment (johnlcox) [reply health/service/src/main/java/com/cerner/beadledom/health/internal/HealthChecker.java]: Existing tests cover the other refactorings I did, but I don't know how to test a log message.

Comment (johnlcox) [reply health/service/src/main/java/com/cerner/beadledom/health/internal/HealthChecker.java]: The [slf4j-test](http://projects.lidalia.org.uk/slf4j-test) project looks interesting for this, but we will lose our regular log output during tests if we use it.

Comment (nathanschile) [reply health/service/src/main/java/com/cerner/beadledom/health/internal/HealthChecker.java]: don't worry about it.

**Pull request merged**